keyboards/kinesis/keymaps/heatxsink/keymap.c

```diff
@@ -0,0 +1,247 @@
+        _______,KC_P7,  KC_P8,  KC_P9,  KC_PMNS,_______,
+        _______,KC_P4,  KC_P5,  KC_P6,  KC_PPLS,_______,
+        _______,KC_P1,  KC_P2,  KC_P3,  KC_PENT,_______,
+                KC_LPRN,KC_RPRN,KC_PDOT,KC_PENT,
+        // RTHUMB
+        _______,_______,
+        _______,
+        _______,KC_PENT,KC_P0
+    ),
+
+    [_LAYERS] = LAYOUT (
+        // LHAND
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,QWERTY ,CMMDH  ,WM     ,KEYPAD ,_______,
+        _______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,
+                _______,_______,_______,_______,
+        // LTHUMB
+                _______,_______,
+                        _______,
+        _______,_______,_______,
+
+        // RHAND
+        _______,_______,_______,_______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,
+        _______,_______,_______,_______,_______,_______,
+                _______,_______,_______,_______,
+        // RTHUMB
+        _______,_______,
+        _______,
+        _______,_______,_______
+    ),
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    switch (get_highest_layer(state)) {
+        case _QWERTY:
+            // Caps Lock Lit
+            writePinLow(LED_CAPS_LOCK_PIN);
+            writePinHigh(LED_NUM_LOCK_PIN);
+            writePinHigh(LED_SCROLL_LOCK_PIN);
+            writePinHigh(LED_COMPOSE_PIN);
+            break;
+        case _COLEMAK_MOD_DH:
+            writePinHigh(LED_CAPS_LOCK_PIN);
+            // Num Lock Lit
+            writePinLow(LED_NUM_LOCK_PIN);
+            writePinHigh(LED_SCROLL_LOCK_PIN);
+            writePinHigh(LED_COMPOSE_PIN);
+            break;
+        case _WORKMAN:
+            writePinHigh(LED_CAPS_LOCK_PIN);
+            writePinHigh(LED_NUM_LOCK_PIN);
+            // Scroll Lock Lit
+            writePinLow(LED_SCROLL_LOCK_PIN);
+            writePinHigh(LED_COMPOSE_PIN);
+
+            break;
+        case _KEYPAD:
+            writePinHigh(LED_CAPS_LOCK_PIN);
+            writePinHigh(LED_NUM_LOCK_PIN);
+            writePinHigh(LED_SCROLL_LOCK_PIN);
+            // Compose Lit
+            writePinLow(LED_COMPOSE_PIN);
+            break;
+        case _LAYERS:
+            // All LEDs Lit
+            writePinLow(LED_SCROLL_LOCK_PIN);
+            writePinLow(LED_NUM_LOCK_PIN);
+            writePinLow(LED_CAPS_LOCK_PIN);
+            writePinLow(LED_COMPOSE_PIN);
+            break;
+        default:
+            // Default layer
+            writePinHigh(LED_NUM_LOCK_PIN);
+            writePinHigh(LED_SCROLL_LOCK_PIN);
+            writePinHigh(LED_COMPOSE_PIN);
+            writePinHigh(LED_CAPS_LOCK_PIN);
+            break;
```

Comment:
From a simplicity standpoint, you could write high to start with, and then write low only for the pins that need to be.  

```suggestion
    writePinHigh(LED_NUM_LOCK_PIN);
    writePinHigh(LED_SCROLL_LOCK_PIN);
    writePinHigh(LED_COMPOSE_PIN);
    writePinHigh(LED_CAPS_LOCK_PIN);
    switch (get_highest_layer(state)) {
        case _QWERTY:
            // Caps Lock Lit
            writePinLow(LED_CAPS_LOCK_PIN);
            break;
        case _COLEMAK_MOD_DH:
            // Num Lock Lit
            writePinLow(LED_NUM_LOCK_PIN);
            break;
        case _WORKMAN:
            // Scroll Lock Lit
            writePinLow(LED_SCROLL_LOCK_PIN);
            break;
        case _KEYPAD:
            // Compose Lit
            writePinLow(LED_COMPOSE_PIN);
            break;
        case _LAYERS:
            // All LEDs Lit
            writePinLow(LED_SCROLL_LOCK_PIN);
            writePinLow(LED_NUM_LOCK_PIN);
            writePinLow(LED_CAPS_LOCK_PIN);
            writePinLow(LED_COMPOSE_PIN);
            break;
```

This isn't required, but may be a bit cleaner.

Reply:
Thank you for pointing this out ... I dig cleaner!